nixnet/types.py

```diff
@@ -361,14 +371,27 @@ def __repr__(self):
         # type: () -> typing.Text
         """CanFrame debug representation.
 
-        >>> CanFrame(1, constants.FrameType.CAN_DATA)
-        CanFrame(CanIdentifier(0x1), echo=False, type=FrameType.CAN_DATA, timestamp=0x0, payload=...)
+        >>> CanFrame(1)
+        CanFrame(CanIdentifier(0x1))
+        >>> CanFrame(1, constants.FrameType.CANFD_DATA, b'\x01')
+        CanFrame(CanIdentifier(0x1), type=FrameType.CANFD_DATA, len(payload)=1)
         """
-        return "CanFrame({}, echo={}, type={}, timestamp=0x{:x}, payload=...)".format(
+        optional = []
+        if self.echo:
+            optional.append('echo={}'.format(self.echo))
+        if self.type != constants.FrameType.CAN_DATA:
+            optional.append('type={}'.format(self.type))
+        if self.timestamp != 0:
+            optional.append('timestamp=0x{:x}'.format(self.timestamp))
+        if self.payload:
+            optional.append('len(payload)={}'.format(len(self.payload)))
+        if optional:
+            optional_params = ', {}'.format(", ".join(optional))
+        else:
+            optional_params = ''
+        return "CanFrame({}{})".format(
             self.identifier,
-            self.echo,
-            self.type,
-            self.timestamp)
+            optional_params)
```

Comment:
I've been going back and forth on the optional.

As long as payload only shows the length, we might be able to get away without the optional parameters.  

The value of hiding the default fields is it makes meaningful information standout more.  My only concern is alignment between rows of frames being off, making it harder to spot specific differences (a missing field will be obvious of course).  I'm assuming that when comparing frames, they will most likely be similar to each other wrt which fields are optional, so they'll probably still line up.